rust/src/dataset.rs

```diff
@@ -126,6 +126,12 @@ impl Dataset {
         Self::checkout_manifest(object_store, base_path, &manifest_file).await
     }
 
+    pub async fn checkout_version(&self, version: u64) -> Result<Self> {
```

Comment:
Just `checkout` ?  Also, pls add some doc to the public API

Reply:
`checkout` is already a static method so this needs a new name. Will add docstring